src/main/java/com/codeus/winter/config/impl/BeanDefinitionRegistryImpl.java

```diff
@@ -0,0 +1,49 @@
+package com.codeus.winter.config.impl;
+
+import com.codeus.winter.config.BeanDefinition;
+import com.codeus.winter.config.BeanDefinitionRegistry;
+import com.codeus.winter.exception.BeanDefinitionStoreException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class BeanDefinitionRegistryImpl implements BeanDefinitionRegistry {
```

Comment:
Accordingly, your BeanDefinitionRegistry is implemented as a separate class, not part of BeanFactory. Please fix the relationships/remove inheritance in the following classes:
- DefaultBeanFactory
- AnnotationApplicationContext
- ClassPathBeanDefinitionScanner